redhat-cop/infra-ansible
# Update IDM User Management

### What does this PR do?
This PR adds the ability to delete/disable users and groups via the `state` attribute of the `ipa_user` and `ipa_group` modules. It also moves the hierarchy from `manage_users` to `manage_idm_users` to match the naming convention of other user-management roles. I also condensed group management and adding users to groups down into a single playbook as they were essentially the same playbook except for adding the users. Now `configure_group.yml` takes care of ensuring that a group is created and has the specified users in a single loop versus having seperate playbooks each with their own.

### How should this be tested?

Modify `roles/user-management/manage-idm-users/tests/create_idm.yml` and `roles/user-management/manage-idm-users/tests/inventory` to point to an existing IDM server

`ansible-playbook -i /path-to/infra-ansible/roles/user-management/manage-idm-users/tests/inventory /path-to/infra-ansible/roles/user-management/manage-idm-users/tests/create_idm.yml`

This will create all of the users and groups listed in `vars/idm.json`. The initial run will show that everything is still created as expected. You can then run the following to modify the users and groups that have been created. 

`ansible-playbook -i /path-to/infra-ansible/roles/user-management/manage-idm-users/tests/inventory /path-to/infra-ansible/roles/user-management/manage-idm-users/tests/create_idm.yml -e "@vars/idm_mod.json"`

`vars./idm_mod.json` contains the same users/groups, but modifies some of them to be removed and also changes group membership. Once you run this, you should see that one of the users has been removed and that group membership of two of the groups has been changed/removed.

### People to notify
cc: @redhat-cop/infra-ansible
